src/script/cpp_api/s_client.cpp

```diff
@@ -224,6 +224,27 @@ bool ScriptApiClient::on_item_use(const ItemStack &item, const PointedThing &poi
 	return lua_toboolean(L, -1);
 }
 
+bool ScriptApiClient::on_inventory_open(Inventory *inventory)
+{
+	SCRIPTAPI_PRECHECKHEADER
+
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "registered_on_inventory_open");
+
+	std::vector<const InventoryList*> lists = inventory->getLists();
+	std::vector<const InventoryList*>::iterator iter = lists.begin();
+	lua_createtable(L, 0, lists.size());
```

Comment:
either add the code for pushing an inventory to lua to `c_content.cpp` and document this in `client_lua_api` or modify `l_inventory.cpp` to work on the client.